Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -115,11 +110,25 @@ import BraintreeCore
         ]
     }
 
-    func completionHandler(
-        onMainThreadWithPaymentRequest paymentRequest: PKPaymentRequest?,
-        error: Error?,
+    // MARK: - Analytics Completion Helpers
+
+    private func completionWithAnalytics(
+        for analyticsResult: BTAnalyticsResult,
+        nonce: BTApplePayCardNonce? = nil,
+        error: Error? = nil,
+        completion: @escaping (BTApplePayCardNonce?, Error?) -> Void
+    ) {
+        apiClient.sendAnalyticsEvent("apple-pay:tokenize:\(analyticsResult == .success ? "succeeded" : "failed")")
```

Comment:
I am in favor of having our analytic events be searchable as raw strings in the SDKs, instead of the concat logic making them harder to search / visualize.

Reply:
Also, if we stick _did_ with the enum - we could give the cases associated String values which would simplify the ternary operator logic, as well as make sure we have consistency with the naming of "succeeded" "failed" "canceled" spelling and tense.

Reply:
I think that's fine if we feel it'd be more helpful to have it be more searchable as the raw string. I think we can still use 1 method to capture `success/failure` vs having 2. Something like this in this method would achieve both:
```
if analyticsResult == .success {
    apiClient.sendAnalyticsEvent("apple-pay:tokenize:succeeded")
} else {
    apiClient.sendAnalyticsEvent("apple-failed")
}

completion(nonce, error)
```

Agree that having the `stringValue` on the enum case would be cleaner if we want to go that route. I personally feel that it's more important to ensure we are using the same tense across all modules over the search-ability for the raw string, but happy to do that as well if folks feel strongly about that aspect. With the string values this could be simplified to:
```
apiClient.sendAnalyticsEvent("apple-pay:tokenize:\(analyticsResult.stringValue)")
```

Thoughts?

Reply:
I like that idea of using associated String values. So are we adding a third option of "canceled"? 
That option is only applicable for some cases, not most, right?